apps/emqx_durable_storage/src/emqx_ds_storage_layer.erl

```diff
@@ -135,9 +143,10 @@
 %% API funcions
 %%================================================================================
 
--spec start_link(emqx_ds:shard(), emqx_ds_storage_layer:options()) -> {ok, pid()}.
-start_link(Shard, Options) ->
-    gen_server:start_link(?REF(Shard), ?MODULE, {Shard, Options}, []).
+-spec start_link(emqx_ds:shard(), emqx_ds:keyspace(), emqx_ds_storage_layer:options()) ->
+    {ok, pid()}.
+start_link(Shard, Keyspace, Options) ->
+    gen_server:start_link(?REF(Shard), ?MODULE, {Shard, Keyspace, Options}, []).
```

Comment:
Q: Shouldn't `Keyspace` also be part of process name here?

Reply:
Since each shard belongs to a single keyspace, I was thinking that it might not be needed.

But if we assume that shards from different keyspaces might use the same name, then that's a good idea.

Reply:
Now I'm thinking if keyspace also shouldn't be an argument to `storage_layer:store/5`, if shard ids may collide.  :thinking:

Reply:
Might be worth exploring making it part of the _shard_ type instead? I.e. `-type shard() :: {_Keyspace, _ShardID}.`.